opentelemetry-api/src/opentelemetry/trace/sampling.py

```diff
@@ -12,6 +12,55 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+"""
+Trace data is often produced in large volumes, it is not only expensive to collect and store but also expensive to transmit.
+
+In order to strike a balance between observability and expenses, traces are sampled. Sampling is the process by which a decision is made on whether to process/export a span or not.
+
+OpenTelemetry provides two types of samplers:
```

Comment:
I might consider linking to the spec instead of explaining the samplers in full. There's a lot of redundant information with the spec here.

Reply:
fair enough, added a link to spec instead